### ISSUE TITLE
Allow passing of the measurements as CSV file

### ISSUE THREAD

**Issue by fedorov** [QIICR/dcmqi#165]
The idea is to simplify the process of creating the TID1500 SRs by allowing the user to pass two CSV files that summarize the measurement. One with the actual measurements, where each row is indexed by the finding name, and each column is indexed by a unique quantity/unit code. The second CSV would have the first column corresponding to the quantity/unit code from the first spreadsheet, and the other columns corresponding to the code tuples that are needed to describe each of the measurements.

**Comment by pieper** [comment]
Is this really simpler to create than the json?  Is there software that
already generates data in this format?

**Comment by fedorov** [comment]
I think it is a lot simpler, because CSV is what people usually use to store this kind of information. If one has to code creation of that file, it probably doesn't matter. But for the situations dealing with the legacy data, I think it is more straightforward. We discussed this topic with @lassoan and @kalpathy at the project week. This idea comes from Jayashree, since this is what they used in some existing project. @lassoan could add support of this for exporting Slicer tables. Of course, we could support output of any format in Slicer, but JSON would only be useful for dcmqi, and CSV export is multipurpose and has to be supported no matter what.

**Comment by pieper** [comment]
As long as people find it useful it's okay with me.

**Comment by fedorov** [comment]
https://github.com/ben-strasser/fast-cpp-csv-parser

**Comment by che85** [comment]
@fedorov The mentioned library makes use of C++11 features. Do we want that?

**Comment by fedorov** [comment]
I saw a mention, but was not sure if it really depends on it. No, we don't want it if it is mandatory. I just put the link here for the reference. Will need to find something else then.

**Comment by lassoan** [comment]
VTK master now requires C++11, so it may not be a blocking issue.
VTK already has a csv parser (that's what we use now to read/write table nodes), but it's not very smart (cannot handle escape characters, requires the first row to be non-empty, etc). I think the best would be to make VTK's csv reader/writer smarter - potentially by integrating a small parser like this fast-cpp-csv-parser.

**Comment by fedorov** [comment]
dcmqi does not depend on VTK, so that approach won't work for us.

I would definitely prefer focusing on a standalone CSV library, and then integrate it with VTK. There a lot more people who need to deal with CSV than with visualization.

**Comment by lassoan** [comment]
OK, if dcmqi does not depend on VTK yet then certainly we shouldn't introduce that dependency.

**Comment by lassoan** [comment]
Actually, it's good if we evaluate this parser independently and if it works well then we'll propose to use it in VTK.

**Comment by fedorov** [comment]
Yes, right. But keeping dcmqi out of the picture, I still think it would be nice for VTK to delegate CSV handling to an external library. Otherwise duplication of effort is inevitable.

> Actually, it's good if we evaluate this parser independently and if it works well then we'll propose to use it in VTK.

I want to keep compatibility with Slicer, so dependency on C++11 is a real blocker. We can't consider it.

**Comment by lassoan** [comment]
As VTK master now requires C++11, Slicer will require C++11 soon.

**Comment by dzenanz** [comment]
@fedorov is there any progress on this?

**Comment by fedorov** [comment]
No, not yet @dzenanz ...
